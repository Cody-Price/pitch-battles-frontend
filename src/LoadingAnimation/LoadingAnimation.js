import React, { Component } from "react";
import "./LoadingAnimation.css";

class LoadingAnimation extends Component {
  render() {
    return <div className="loading-animation" />;
  }
}

export default LoadingAnimation;
