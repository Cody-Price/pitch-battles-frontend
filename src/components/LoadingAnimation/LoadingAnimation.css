.loading-animation-wrapper {
  position: relative;
  margin: 4rem auto;
  width: 152px;
}

.loading-animation {
  width: 150px;
  height: 150px;
  position: relative;
  overflow: hidden;
  background: url(./images/cloudbg.png);
  background-size: cover;
  border-radius: 50%;
}

.circle-border {
  height: 152px;
  width: 152px;
  left: -1px;
  top: -1px;
  background: url(./images/circle.png);
  background-size: cover;
  image-rendering: pixelated;
  z-index: 1000;
  position: absolute;
}

.duck {
  height: 48px;
  width: 48px;
  position: absolute;
  top: 40px;
  left: 50px;
  animation: duck 3s linear infinite;
  z-index: 1000;
}

@keyframes duck {
  0% {
    background: url(./images/duck1.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  80% {
    background: url(./images/duck1.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  80.01% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  81% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  81.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  82% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  82.01% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  83% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  83.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  84% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  84.01% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  85% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  85.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  86% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  86.01% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  87% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  88.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  88% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  89.01% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  90% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  90.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  91% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  91.01% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  92% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  92.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  93% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  93.01% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  94% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  94.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  95% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  95.01% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  96% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  96.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  97% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  97.01% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  98% {
    background: url(./images/duck2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  98.01% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  99% {
    background: url(./images/duck3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  99.01% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  100% {
    background: url(./images/duck4.png);
    background-size: cover;
    image-rendering: pixelated;
  }
}

.walkway {
  width: 450px;
  height: 150px;
  background: url(./images/walkway.png);
  position: absolute;
  background-size: cover;
  image-rendering: pixelated;
  z-index: 4;
  margin-top: 0.5rem;
}

.walkway.true {
  animation: walkway 8s linear infinite;
}

@keyframes walkway {
  0% {
    left: 0px;
    top: 0;
  }
  100% {
    left: -300px;
  }
}

.cat {
  position: absolute;
  top: 36px;
  left: 52px;
  animation: catwalk 1s linear infinite;
  height: 48px;
  width: 45px;
  z-index: 5;
}

@keyframes catwalk {
  0% {
    background: url(./images/left1.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  25% {
    background: url(./images/left1.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  25.01% {
    background: url(./images/left2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  50% {
    background: url(./images/left2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  50.01% {
    background: url(./images/left3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  75% {
    background: url(./images/left3.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  75.01% {
    background: url(./images/left2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
  100% {
    background: url(./images/left2.png);
    background-size: cover;
    image-rendering: pixelated;
  }
}

.cloud1 {
  position: absolute;
  width: 450px;
  height: 150px;
  background: url("./images/cloud1.png");
  background-size: cover;
  image-rendering: pixelated;
  animation: cloud1 12s linear infinite;
}

@keyframes cloud1 {
  0% {
    left: 100px;
    top: 0px;
  }
  100% {
    left: -300px;
    top: 0px;
  }
}

.cloud2 {
  position: absolute;
  width: 450px;
  height: 150px;
  background: url("./images/cloud2.png");
  background-size: cover;
  image-rendering: pixelated;
  animation: cloud1 20s linear infinite;
}

@keyframes cloud2 {
  0% {
    left: 0;
    top: 10px;
  }
  100% {
    left: -300px;
    top: 10px;
  }
}
