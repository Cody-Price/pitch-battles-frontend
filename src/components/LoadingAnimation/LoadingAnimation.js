import React, { Component } from "react";
import "./LoadingAnimation.css";

class LoadingAnimation extends Component {
  render() {
    const animals = {
      duck: false,
      cat: true
    };
    return (
      <section className="loading-animation-wrapper">
        <div className="circle-border" />

        <div className="loading-animation">
          <div className={this.props.animal} />
          <div className={`walkway ${animals[this.props.animal]}`} />
          <div className="cloud1" />
          <div className="cloud2" />
        </div>
      </section>
    );
  }
}

export default LoadingAnimation;
