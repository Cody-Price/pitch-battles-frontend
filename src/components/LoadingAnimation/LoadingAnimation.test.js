import React from "react";
import LoadingAnimation from "./LoadingAnimation";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("LoadingAnimation", () => {
  let wrapper;
  const mockTime = [1];

  beforeEach(() => {
    wrapper = shallow(<LoadingAnimation time={mockTime} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
