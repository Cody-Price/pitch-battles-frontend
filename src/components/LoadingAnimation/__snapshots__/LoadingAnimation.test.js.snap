// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingAnimation should match the snapshot 1`] = `
<section
  className="loading-animation-wrapper"
>
  <div
    className="circle-border"
  />
  <div
    className="loading-animation"
  >
    <div />
    <div
      className="walkway undefined"
    />
    <div
      className="cloud1"
    />
    <div
      className="cloud2"
    />
  </div>
</section>
`;
